pydantic/types.py

```diff
@@ -339,13 +340,33 @@ def __modify_schema__(cls, field_schema: Dict[str, Any]) -> None:
             maximum=cls.le,
             multipleOf=cls.multiple_of,
         )
+        cls._schema_ieee_compatibility_transform(field_schema)
 
     @classmethod
     def __get_validators__(cls) -> 'CallableGenerator':
         yield strict_float_validator if cls.strict else float_validator
         yield number_size_validator
         yield number_multiple_validator
 
+    @classmethod
+    def _schema_ieee_compatibility_transform(cls, field_schema: Dict[Any, Any]) -> None:
```

Comment:
is there a good reason this is a separate method?

Reply:
It does not need to be. I'll merge it with `__modify_schema__`.